assets/js/base/components/price-slider/index.js

```diff
@@ -0,0 +1,405 @@
+
+	useEffect( () => {
+		if (
+			value.min === undefined ||
+			value.max === undefined ||
+			min > value.min ||
+			max < value.max
+		) {
+			setValue( {
+				min,
+				max,
+			} );
+		}
+		setFormattedValue( {
+			min: formatCurrencyForInput(
+				value.min,
+				priceFormat,
+				currencySymbol
+			),
+			max: formatCurrencyForInput(
+				value.max,
+				priceFormat,
+				currencySymbol
+			),
+		} );
+	}, [ min, max ] );
+
+	useEffect( () => {
+		setFormattedValue( {
+			min: formatCurrencyForInput(
+				value.min,
+				priceFormat,
+				currencySymbol
+			),
+			max: formatCurrencyForInput(
+				value.max,
+				priceFormat,
+				currencySymbol
+			),
+		} );
+	}, [ value ] );
+
+	useEffect( () => {
+		if ( ! showFilterButton && ! isLoading ) {
+			triggerChangeCallback();
+		}
+	}, [ debouncedChangeValue ] );
+
+	const triggerChangeCallback = () => {
+		onChange( value );
+	};
+
+	/**
+	 * Handles styles for the shaded area of the range slider.
+	 */
+	const getProgressStyle = () => {
+		const low =
+			Math.round( 100 * ( ( value.min - min ) / ( max - min ) ) ) - 0.5;
+		const high =
+			Math.round( 100 * ( ( value.max - min ) / ( max - min ) ) ) + 0.5;
+
+		return {
+			'--low': low + '%',
+			'--high': high + '%',
+		};
+	};
+
+	/**
+	 * Works around an IE issue where only one range selector is visible by changing the display order
+	 * based on the mouse position.
+	 *
+	 * @param {obj} event event data.
+	 */
+	const findClosestRange = ( event ) => {
+		if ( isLoading ) {
+			return;
+		}
+		const bounds = event.target.getBoundingClientRect();
+		const x = event.clientX - bounds.left;
+		const minWidth = minRange.current.offsetWidth;
+		const minValue = minRange.current.value;
+		const maxWidth = maxRange.current.offsetWidth;
+		const maxValue = maxRange.current.value;
+
+		const minX = minWidth * ( minValue / max );
+		const maxX = maxWidth * ( maxValue / max );
+
+		const minXDiff = Math.abs( x - minX );
+		const maxXDiff = Math.abs( x - maxX );
+
+		/**
+		 * The default z-index in the stylesheet as 20. 20 vs 21 is just for determining which range
+		 * slider should be at the front and has no meaning beyond
+		 */
+		if ( minXDiff > maxXDiff ) {
+			minRange.current.style.zIndex = 20;
+			maxRange.current.style.zIndex = 21;
+		} else {
+			minRange.current.style.zIndex = 21;
+			maxRange.current.style.zIndex = 20;
+		}
+	};
+
+	/**
+	 * Called when the slider is dragged.
+	 * @param {obj} event Event object.
+	 */
+	const onDrag = ( event ) => {
+		const isMin = event.target.classList.contains(
+			'wc-block-price-filter__range-input--min'
+		);
+		const values = constrainRangeSliderValues(
+			[ minRange.current.value, maxRange.current.value ],
+			min,
+			max,
+			step,
+			isMin
+		);
+		setValue( {
+			min: parseInt( values[ 0 ], 10 ),
+			max: parseInt( values[ 1 ], 10 ),
+		} );
+	};
+
+	/**
+	 * Called when a price input loses focus.
+	 * @param {obj} event Event object.
+	 */
+	const onInputBlur = ( event ) => {
```

Comment:
This should be wrapped in `useCallback` along with the appropriate dependencies because it is an event handler passed through to a component.